File: 0443-string-compression/0443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        read=0
        write=0
        
        while read < len(chars):
            char = chars[read]
            count = 0
            
            while read<len(chars) and char == chars[read]:
                read += 1
                count += 1
                
            chars[write] = char
            write += 1
            if count > 1:
                for digit in str(count):
                    chars[write] = digit
                    write += 1
 
        return write

File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int read=0;
        int write=0;
        while(read<chars.size()){
            int count=0;
            char c=chars[read];
            while(read<chars.size() && c==chars[read]){
                count++;
                read++;
            }
            chars[write]=c;
            write++;
            if(count>1){
                for (char digit : to_string(count)) {
                    chars[write]=digit;
                    write++;
                }
            }
        }
        return write;
    }
};